src/main/java/com/example/onlinebookstore/controller/AuthenticationController.java

```diff
@@ -0,0 +1,32 @@
+package com.example.onlinebookstore.controller;
+
+import com.example.onlinebookstore.dto.UserDto;
+import com.example.onlinebookstore.dto.UserRegistrationRequestDto;
+import com.example.onlinebookstore.exception.RegistrationException;
+import com.example.onlinebookstore.service.UserService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.AllArgsConstructor;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Users", description = "Endpoints for managing users")
+@RestController
+@AllArgsConstructor
+@RequestMapping("/api/auth")
```

Comment:
```suggestion
@RequestMapping("/auth")
```
add `server.servlet.context-path=/api` to application.properties